recipes/hdf5/all/conanfile.py

```diff
@@ -118,6 +120,10 @@ def generate(self):
         cmakedeps.generate()
 
         tc = CMakeToolchain(self)
+        if not can_run(self):
+            c = self.conf.get("user.qemu:path")
```

Comment:
you can inject CMAKE_CROSSCOMPILING_EMULATOR without changing recipe through user_toolchain

in profile:
[conf]
hdf5*:tools.build.cross_building:can_run=true
tools.cmake.cmaketoolchain:user_toolchain=[".....path_to_toolchain.cmake"]

in toolchain:
set(CMAKE_CROSSCOMPILING_EMULATOR ...

Or with more modern cmake (3.28+) you can even use environment variable through profile instead of toolchain

Reply:
Great Idea, unfortunately in the recipe will be still required, since it is currently hardwired to fail in a cross-build environment ([conanfile.py](https://github.com/conan-io/conan-center-index/blob/35d081612727c323cf5b8b698fe68dcb413353da/recipes/hdf5/all/conanfile.py#L88-L90)).
To solve this issue, we have three options in my eyes:
1. We could either blindly allow cross compilation and accept the fact the builds will fail when crossbuilding.
2. With you proposed alternative, we must check if CMAKE_CROSSCOMPILING_EMULATOR is set in the toolchain file (At least in my case is CMake 3.28 not available :-/). 
3. Like it is currently implemented, we add an option to the recipe.
What is the best option moving forward in your opinion?

Reply:
> Great Idea, unfortunately in the recipe will be still required, since it is currently hardwired to fail in a cross-build environment ([conanfile.py](https://github.com/conan-io/conan-center-index/blob/35d081612727c323cf5b8b698fe68dcb413353da/recipes/hdf5/all/conanfile.py#L88-L90)). To solve this issue, we have three options in my eyes:
> 
>     1. We could either blindly allow cross compilation and accept the fact the builds will fail when crossbuilding.
> 
>     2. With you proposed alternative, we must check if CMAKE_CROSSCOMPILING_EMULATOR is set in the toolchain file (At least in my case is CMake 3.28 not available :-/).
> 
>     3. Like it is currently implemented, we add an option to the recipe.
> 
> 
> What is the best option moving forward in your opinion?

With hdf5*:tools.build.cross_building:can_run=true it will not fail in this check

Reply:
I've just shared how I am currently using this recipe with emscripten (hdf5 1.14.1).

Reply:
Sorry for bothering you again. I cannot get your approach to work with a conanfile.py. 
Currently I use:
```Python
 def requirements(self):
        self.requires("hdf5/1.14.3")
        self.options["hdf5:tools.options.cross_building"].can_run = True
```
Unfortunately, this does not change anything, and the build still fails with:
`hdf5/1.14.3: Invalid ID: Current recipe doesn't support cross-building (yet)`

Do you have any idea what I am missing?